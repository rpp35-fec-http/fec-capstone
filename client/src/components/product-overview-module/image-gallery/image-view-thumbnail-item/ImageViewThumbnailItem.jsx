import React from 'react';
import './imageViewThumbnailItem.scss';
import PendingIcon from '@mui/icons-material/Pending';
import DefaultImage from '../../../pages/home/product-carousel/product-card/Image-coming-soon.svg';

const ImageViewThumbnailItem = ({
   image,
   index,
   currentIndex,
   handleThumbnailClick,
   isEnlargedView,
}) => {
   const handleClick = () => {
      handleThumbnailClick(index);
   };

   let currentClass;
   if (index !== currentIndex) {
      currentClass = 'thumbnail-default';
   } else {
      currentClass = 'thumbnail-active';
   }

   return (
      <div
         className={currentClass}
         onClick={handleClick}
         data-testid={`thumbnail-img-${index}`}
      >
         {isEnlargedView ? (
            <PendingIcon color='primary' />
         ) : (
            <img
               className='thumbnail-img'
               alt='thumbnail image'
               src={image || DefaultImage}
            ></img>
         )}
      </div>
   );
};

export default ImageViewThumbnailItem;
