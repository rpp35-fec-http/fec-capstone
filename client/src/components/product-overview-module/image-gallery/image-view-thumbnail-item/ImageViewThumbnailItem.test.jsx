import renderer from 'react-test-renderer';
import ImageViewThumbnailItem from './ImageViewThumbnailItem.jsx';

it('renders correctly', () => {
  const tree = renderer.create(<ImageViewThumbnailItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
