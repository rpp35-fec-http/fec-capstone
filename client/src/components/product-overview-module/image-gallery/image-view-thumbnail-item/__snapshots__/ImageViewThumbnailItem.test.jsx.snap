// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="thumbnail-active"
  data-testid="thumbnail-img-undefined"
  onClick={[Function]}
>
  <img
    alt="thumbnail image"
    className="thumbnail-img"
    src="test-file-stub"
  />
</div>
`;
